pkg/controller/catalog/server/controller.go

```diff
@@ -52,123 +52,149 @@ func createController(s storage.Storage) (*controller, error) {
 	}, nil
 }
 
-func (c *controller) updateServiceInstance(in *scmodel.ServiceInstance) error {
+func (c *controller) updateServiceInstance(in *servicecatalog.Instance) error {
 	// Currently there's no difference between create / update,
 	// but for prepping for future, split these into two different
 	// methods for now.
 	return c.createServiceInstance(in)
 }
 
-func (c *controller) createServiceInstance(in *scmodel.ServiceInstance) error {
-	broker, err := storage.GetBrokerByServiceClass(c.storage, in.ServiceID)
+func (c *controller) createServiceInstance(in *servicecatalog.Instance) error {
+	broker, err := storage.GetBrokerByServiceClass(c.storage.Brokers(), c.storage.ServiceClasses(), in.Spec.CFServiceID)
```

Comment:
Makes total sense... I actually think this is a mistake here. I'm going to fix that here and now instead of in a follow-up.